### PR TITLE
A better i18n support

### PR DESCRIPTION
**Pull request by moesoha**
**TL;DR**

What's included in this PR:
  - A better locale module
  - Support localize texts in WebView pages
  - Modified some(not all) extension codes for supporting i18n

What's NOT includes:
  - Full Chinese support

All translations added to the `./locale/` are for developing this feature. But I'll bring full Chinese support later _(:3

----------

Due to the inconvenience of using `vscode-nls` in this project, I implemented a new module for i18n (`./src/locale.ts`). With this new module, we can easily to localize this extension and its WebView pages.

For example in the extension, with `import { localize as i18n } from './src/locale.ts'` we can just write `i18n('message', 'Hello, {0}.', username)` to produce `Hello, Soha.` for English and `你好，Soha。` for Chinese. `'message'` is the translation domain, the second argument is the message, and the following arguments will be inserted into message after it gets translated.

I also wrote a loader for WebView pages. just put `data-i18n-trans` in tags, it will translate its `textContent` once page loaded. Or use `window.i18nTrans` is also fine.

I chose YAML for the localization file because compared to JSON, it costs less time while writing localization files.

### CODE REVIEW

**Comment by moesoha** [comment]
@ThadHouse PTAL

File: vscode-wpilib/package.json [hunk 1]
```diff
@@ -462,4 +465,4 @@
         "wpilib-riolog": "^0.2.0",
         "xml2js": "^0.4.19"
     }
-}
+}
```
Comment (ThadHouse) [inline vscode-wpilib/package.json]: The EOL at EOF needs to not be removed from this.

File: vscode-wpilib/resources/webviews/projectcreator.html [hunk 2]
```diff
@@ -17,39 +17,41 @@
...
+  <div id="teamnumberdiv" data-i18n-trans="ui">Enter a team number</div>
   <input id="teamNumber" type="text" style="width: 100%; margin-top: 10px; margin-bottom: 10px"
   />
   <br/>
   <div id="desktopdiv">
-    This is needed for simulation and unit testing support, however there are some cases where this will do
-    some unexpected things during build. In addition, not all vendor libraries support desktop.
-    This option can be set with the command "WPILib: Set Desktop Support" at any time.
+	<span data-i18n-trans data-i18n-key="hintDesktopSupport">
+      This is needed for simulation and unit testing support, however there are some cases where this will do
```
Comment (ThadHouse) [inline vscode-wpilib/resources/webviews/projectcreator.html]: image

For some reason, this is just showing `hintDesktopSupport` rather then the actual text.

File: vscode-wpilib/package.json (hunk 1 above)
Comment (moesoha) [reply vscode-wpilib/package.json]: It was removed by npm, fixed.

**Comment by ThadHouse** [comment]
I want to get this merged so I don't have to worry about it getting out of date, but I would like a doc on how to use this. We have a `TRANSLATING.md` file, can you write up some usage docs in that in a separate PR?

Other then that, looks good to me.

**Comment by moesoha** [comment]
Thanks. I'll write docs for it a few days later.
